framework/decode/vulkan_replay_dump_resources_transfer.h

```diff
@@ -0,0 +1,826 @@
+                                      const VulkanAccelerationStructureKHRInfo* d,
+                                      VkCopyAccelerationStructureModeKHR        m,
+                                      const graphics::VulkanDeviceTable&        dt,
+                                      const CommonObjectInfoTable&              oit,
+                                      const VulkanPerDeviceAddressTrackers&     at) :
+                src_as(s),
+                dst_as(d->capture_id), mode(m), vk_objects(dt, oit, at, d)
+            {}
+
+            format::HandleId                   src_as;
+            format::HandleId                   dst_as;
+            VkCopyAccelerationStructureModeKHR mode;
+            CopiedAccelerationStructure        vk_objects;
+        };
+
+        // TransferParams constructors, one for each transfer command
+        TransferParams() = delete;
+
+        // kInitBufferCommand
+        TransferParams(format::HandleId                   b,
+                       const uint8_t*                     d_p,
+                       uint64_t                           size,
+                       const graphics::VulkanDeviceTable& dt,
+                       const VulkanDeviceInfo*            pdi,
+                       TransferCommandTypes               t) :
+            params(std::in_place_type<InitBufferMetaCommand>, b, d_p, size),
+            type(t), has_before_command(false), device_table_(dt), parent_device_info_(pdi)
+        {
+            GFXRECON_ASSERT(t == TransferCommandTypes::kCmdInitBuffer);
+        }
+
+        // kInitImageCommand
+        TransferParams(format::HandleId                   i,
+                       VkImageAspectFlagBits              a,
+                       const VulkanImageInfo*             img_info,
+                       const graphics::VulkanDeviceTable& dt,
+                       const VulkanDeviceInfo*            pdi,
+                       TransferCommandTypes               t) :
+            params(std::in_place_type<InitImageMetaCommand>, i, a, img_info),
+            type(t), has_before_command(false), device_table_(dt), parent_device_info_(pdi)
+        {
+            GFXRECON_ASSERT(t == TransferCommandTypes::kCmdInitImage);
+        }
+
+        // CmdCopyBuffer
+        TransferParams(format::HandleId                   s,
+                       format::HandleId                   d,
+                       const graphics::VulkanDeviceTable& dt,
+                       const VulkanDeviceInfo*            pdi,
+                       bool                               bc,
+                       TransferCommandTypes               t) :
+            params(std::in_place_type<CopyBuffer>, s, d),
+            type(t), has_before_command(bc), device_table_(dt), parent_device_info_(pdi)
+        {
+            GFXRECON_ASSERT(t == TransferCommandTypes::kCmdCopyBuffer);
+
+            if (bc)
+            {
+                before_params = CopyBuffer(s, d);
```

Comment:
It looks like before_params is identical to params; can you help me understand what the purpose is of having separate params and before_params?